module/programs/helper.js

```diff
@@ -1067,6 +1069,46 @@ module.exports = class ProgramsHelper {
 						result: [],
 					})
 				}
+				if (projectsCount) {
+					//Filtering out all the program IDs
+					let allProjectIds = programsData.flatMap((program) => program._id || [])
+					let projectDocuments = await projectQueries.getAggregate([
+						{
+							$match: {
+								programId: { $in: allProjectIds },
+							},
+						},
+						{
+							$group: {
+								_id: '$programId', // Group by programId
+								ongoingCount: {
+									$sum: { $cond: [{ $ne: ['$status', CONSTANTS.common.SUBMITTED_STATUS] }, 1, 0] },
+								},
+								completedCount: {
+									$sum: { $cond: [{ $eq: ['$status', CONSTANTS.common.SUBMITTED_STATUS] }, 1, 0] },
+								},
+							},
+						},
+					])
+					//Adding count to the programData
+					const projectCountsMap = projectDocuments.reduce((acc, project) => {
+						acc[project._id.toString()] = {
+							ongoingCount: project.ongoingCount || 0,
+							completedCount: project.completedCount || 0,
+						}
+						return acc
+					}, {})
+
+					// Merge counts using the map
+					programsData.forEach((program) => {
+						const counts = projectCountsMap[program._id.toString()] || {
```

Comment:
@praveenKDass lets call them ongoingProjects and completedProjects

Reply:
Changed @VISHNUDAS-tunerlabs